smarts/core/sensors.py

```diff
@@ -248,32 +275,17 @@ def observe(sim, agent_id, sensor_state, vehicle) -> Tuple[Observation, bool]:
         if vehicle.subscribed_to_neighborhood_vehicles_sensor:
             neighborhood_vehicles = []
             for nv in vehicle.neighborhood_vehicles_sensor():
-                nv_lane = sim.road_map.nearest_lane(
-                    nv.pose.point, radius=vehicle.length
-                )
+                veh_obs = _make_vehicle_observation(sim.road_map, nv)
                 nv_lane_pos = None
-                if nv_lane:
-                    nv_road_id = nv_lane.road.road_id
-                    nv_lane_id = nv_lane.lane_id
-                    nv_lane_index = nv_lane.index
-                    if vehicle.subscribed_to_lane_position_sensor:
-                        nv_lane_pos = vehicle.lane_position_sensor(nv_lane, nv)
-                else:
-                    nv_road_id = None
-                    nv_lane_id = None
-                    nv_lane_index = None
-                neighborhood_vehicles.append(
-                    VehicleObservation(
-                        id=nv.vehicle_id,
-                        position=nv.pose.position,
-                        bounding_box=nv.dimensions,
-                        heading=nv.pose.heading,
-                        speed=nv.speed,
-                        road_id=nv_road_id,
-                        lane_id=nv_lane_id,
-                        lane_index=nv_lane_index,
-                        lane_position=nv_lane_pos,
+                if (
+                    veh_obs.lane_id is not None
+                    and vehicle.subscribed_to_lane_position_sensor
+                ):
+                    nv_lane_pos = vehicle.lane_position_sensor(
+                        sim.road_map.lane_by_id(veh_obs.lane_id), nv
                     )
+                neighborhood_vehicles.append(
+                    veh_obs._replace(lane_position=nv_lane_pos)
```

Comment:
I just noticed these changes (b/c I had a merge conflict trying to merge these into my own branch).  Even though they don't look like they'll break anything (and of course I've already approved this PR and you've already merged it!), I'm curious why you didn't move the `nv_lane_pos` code into the `_make_vehicle_observations()` function with the rest of the code there.